docker/package/wizard_structure.py

```diff
@@ -715,13 +715,13 @@ def import_key(self, key_mode_query, ledger_app=None):
                         f"Waiting for funds to arrive... (Ctrl + C to choose another option)."
                     )
                     try:
-                        # dry-run delegate registration until it succeeds
                         while True:
                             result = get_proc_output(
                                 f"sudo -u tezos {suppress_warning_text} octez-client {tezos_client_options} "
-                                f"register key {baker_alias} as delegate --dry-run"
+                                f"register key {baker_alias} as delegate"
```

Comment:
Now there is an unnecessary register baker call in the setup wizard tho, in the `register_baker` function, which is called right after this one.

On a related point, maybe this function's name should be changed now BTW

Reply:
Since it unharmful, I don't think we should change it

Reply:
> it became unnecessary only for `generate-fresh-key` option

I'm confused, that's exactly what I'm referring to.

I mean, there is a "register key" call to `tezos-client` in the `register_baker` function (the only one not in `import_key`) which I think should be removed now.

AFAICT `register_baker` is only called in `run_setup` _after_ `import_baker_key`, which already registers the delegate.

In fact, the whole `register_baker` function now seems no longer needed and IMO `import_key` should be renamed to `import_delegate_key` to make clear(er) that delegate registration happens there too.

What do you think, does it make sense to you now?

Reply:
`import-key` registers delegate only for `generate-fresh-key` option. If user choose another option, we still need `register_baker` function. But your idea to merge that two function make sense, so that for every option delegate will be registered after import